Makefile

```diff
@@ -35,18 +35,20 @@ endif
 .PHONY: all
 all: clean test build
 
-$(CMDS):
+$(CMDS): vendor
```

Comment:
Not sure why we are adding the vendor dependency here? Which seems to eventually run `go mod vendor` to fill/create the vendor directory (which is .gitignored)? That seems like quite a slowdown; especailly for those who run `git clean -fdxq` a lot.

Reply:
The builds for this repo are all dependent on the presence of the vendor directory and will fail without it. Running `make vendor` again when the `vendor` dir is already there is essentially a no-op since it won't need to re-download them. From my testing `go mod vendor` will not re-download the entire folder every time; it will first grab them from `~/go/pkg/mod` which is a very fast operation. The presence of this folder is also a requirement for the downstream repos, since everything the project is built from needs to exist in the repo, or at least that's my understanding of why.

Reply:
I would argue this is still vendoring; this PR just gets rid of the directory and the build step has to recreate it each time...

Reply:
I guess my point is, if everything is already in the GOPATH, why are we copying it over?

Reply:
I agree with Todd. I think there is no benefit of in copying into vendor since we have local cache for modules.

> The builds for this repo are all dependent on the presence of the vendor directory and will fail without it.

I think this is because our `Makefile` has `GOFLAGS="-mod=vendor"` which is forcing `go build` to use vendor dir.

Try running this in master:

```
sed -i 's/GO := GOFLAGS="-mod=vendor" go/GO := go/g' ./Makefile
rm -r ./vendor
go mod download
make build
```

Even `go mod download` might not be necessary. I think `go build` without `-mod=vendor` will be to download modules (if not present) and build.

Reply:
I agree with you both here and would prefer to not have the `vendor/` dir at all. But I do want to ask the following, not as a challenge but a genuine question as somebody still not terribly familiar with up/down-streaming processes: If we change the build process for the upstream repo to stop using `vendor/`, will that not make the downstream cherry-picking process more complicated? Since the downstream repo _must_ use the `vendor/` dir for the build (somebody correct me if this is wrong), then aren't we beginning to create a significant divergence between the up and downstream repos? Or is this not what you're recommending?

Reply:
I'm also not super familiar with downstreaming, but I'm happy to look into it together on next week.

Reply:
The vendor directory doesn't exist in the downstream staging directory. But there is a root-level vendor.